flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -647,8 +648,10 @@ private FlinkKafkaProducer(
                 new FlinkKafkaProducer.TransactionStateSerializer(),
                 new FlinkKafkaProducer.ContextStateSerializer());
 
-        this.defaultTopicId = checkNotNull(defaultTopic, "defaultTopic is null");
-
+        checkArgument(
+                !org.apache.flink.util.StringUtils.isNullOrWhitespaceOnly(defaultTopic),
```

Comment:
What about `StringUtils#isNotBlank` of package `org.apache.commons.lang3` ?

Reply:
Thank you for your suggestion, it's good, I modified it